lib/chef/provider/directory.rb

```diff
@@ -64,7 +64,13 @@ def define_resource_requirements
               is_parent_writable = lambda do |base_dir|
                 base_dir = ::File.dirname(base_dir)
                 if ::File.exists?(base_dir)
-                  Chef::FileAccessControl.writable?(base_dir)
+                  if Chef::FileAccessControl.writable?(base_dir)
+                    true
+                  elsif Gem::Version.new(node['platform_version']) >= Gem::Version.new('10.11') && Chef::Util::PathHelper.is_sip_path?(base_dir)
```

Comment:
I was going to try to get the OS  version using RbConfig, but ran into this: #3708 

Any ideas on how to get at this info? I could shell out, but that seems super wasteful in this use case.